source/DataHolder/AMDSGenericFlatArrayDataHolder.cpp

```diff
@@ -11,6 +13,30 @@ AMDSLightWeightGenericFlatArrayDataHolder::~AMDSLightWeightGenericFlatArrayDataH
 {
 }
 
+AMDSDataHolder* AMDSLightWeightGenericFlatArrayDataHolder::operator +(AMDSDataHolder &dataHolder)
+{
+	AMDSLightWeightGenericFlatArrayDataHolder *addInputDataHolder = qobject_cast<AMDSLightWeightGenericFlatArrayDataHolder *>(&dataHolder);
+	if (!addInputDataHolder) {
+		AMDSErrorMon::alert(this, 0, QString("Invalid class type (%1) to do the PLUS operation").arg(dataHolder.metaObject()->className()));
+		return 0;
+	}
+
+	AMDSFlatArray data = dataArray() + addInputDataHolder->dataArray();
+
+	AMDSDataHolder* targetDataHolder = AMDSDataHolderSupport::instantiateDataHolderFromClassName(metaObject()->className());
```

Comment:
nice catch. There will be some memory leak. But I don't have solution so far. We need to rework the whole logic on how to handle the data in AMDSClientDataRequest. I will raise an issue and work on it later.